### Suicides rates

In [32]:
import pandas as pd

PATH_TO_FILE = './interactive_data.csv'

data = pd.read_csv(PATH_TO_FILE)

data.head()

Unnamed: 0         Intent         Gender            Age  \
0           1  None selected  None selected  None selected   
1           2  None selected  None selected  None selected   
2           3  None selected  None selected  None selected   
3           4  None selected  None selected  None selected   
4           5  None selected  None selected  None selected   

                     Race  Deaths  Population  Rate  
0           None selected   33599   316299978  10.6  
1                   White   22079   197369634  11.2  
2                   Black    7765    38896382  20.0  
3                Hispanic    3007    54049078   5.6  
4  Asian/Pacific Islander     442    16315561   2.7

In [4]:
len(data)

540

In [5]:
#Important attributes for selection
attrs = data.columns[1:5]
print(attrs)

Index(['Intent', 'Gender', 'Age', 'Race'], dtype='object')


As we inspected the dataset, we can define dictionary *attributes* which is used for setting the desired filter conditions in every claim.

In [ ]:
attributes = {}

In [6]:
def setDefaultAttributes(data, attributes):
    for data_attribute in data.columns[1:5]:
        attributes[data_attribute]='None selected'

In [7]:
setDefaultAttributes(data, attributes)    
print(attributes)

{'Intent': 'None selected', 'Gender': 'None selected', 'Age': 'None selected', 'Race': 'None selected'}


Next, we define useful functions for future work.

In [8]:
def findNumberOfDeathsWithAttributes(data, attributes):
    return data.loc[(data[list(attributes)] == pd.Series(attributes)).all(axis=1)]['Deaths'].iloc[0]

In [9]:
def calculatePercentage(number1, number2):
    return number1/number2*100

### Claim #1:

Nearly *two-thirds* of gun deaths are *suicides*:

In [10]:
setDefaultAttributes(data, attributes)

all_deaths = findNumberOfDeathsWithAttributes(data, attributes)

print(all_deaths)

33599


In [11]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Suicide'

suicide_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(suicide_deaths)

21058


In [12]:
print(calculatePercentage(suicide_deaths, all_deaths))

62.67448435965356


### Conclusion
Accorting to the database, we see that 62.67 percent of all deaths have suicide as their intent.

### Claim #2:

More than *85 percent* of suicide victims are *male*.

In [22]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Suicide'
attributes['Gender']='Male'

suicide_male_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(suicide_male_deaths)

18162


In [23]:
print(calculatePercentage(male_deaths, suicide_deaths))

86.24750688574414


### Conclusion

Accorting to the database, we see that 86.25 percent of suicide victims are male. 

### Claim #3:

Around *a third* of all gun deaths are *homicides*.

In [15]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Homicide'

homicide_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_deaths)

11726


In [16]:
print(calculatePercentage(homicide_deaths, all_deaths))

34.899848209768145


### Conclusion

Accorting to the database, we see that 34.90 percent of all deaths are homicides. 

### Claim #4:

Around *two-thirds* of homicide victims who are *males* in the *age-group of 15--34* are *black*.

In [17]:
setDefaultAttributes(data, attributes)
attributes['Intent']='Homicide'
attributes['Gender'] = 'Male'
attributes['Age'] = '15 - 34'

homicide_male_mid_age_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_male_mid_age_deaths)

6520


In [18]:
setDefaultAttributes(data, attributes)
attributes['Intent'] = 'Homicide'
attributes['Gender'] = 'Male'
attributes['Age'] = '15 - 34'
attributes['Race'] = 'Black'

homicide_male_mid_age_black_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_male_mid_age_black_deaths)

4312


In [19]:
print(calculatePercentage(homicide_male_mid_age_black_deaths, homicide_male_mid_age_deaths))

66.13496932515336


### Conclusion

Accorting to the database, we see that 66.13 percent of homicide victims who are males in the age-group of 15-34 are black. 

### Claim #5:

*Women* constitue only *15 percent* of the total *homicide* victims.

In [20]:
setDefaultAttributes(data, attributes)
attributes['Intent'] = 'Homicide'
attributes['Gender'] = 'Female'

homicide_female_deaths = findNumberOfDeathsWithAttributes(data, attributes)
print(homicide_female_deaths)

1791


In [21]:
print(calculatePercentage(homicide_female_deaths, homicide_deaths))

15.273750639604296


### Conclusion

Accorting to the database, we see that 15.27 percent of homicide victims are female. 